client/my-sites/hosting/cache-card/index.js

```diff
@@ -8,19 +14,90 @@ import { clearWordPressCache } from 'calypso/state/hosting/actions';
 import getRequest from 'calypso/state/selectors/get-request';
 import { shouldRateLimitAtomicCacheClear } from 'calypso/state/selectors/should-rate-limit-atomic-cache-clear';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { useClearEdgeCacheMutation } from './use-clear-edge-cache';
+import { useEdgeCacheQuery } from './use-edge-cache';
+import { useToggleEdgeCacheMutation } from './use-toggle-edge-cache';
 
-const CacheCard = ( {
+const CardBody = styled.div( {
+	display: 'flex',
+	flexDirection: 'column',
+} );
+
+const ToggleContainer = styled.div( {
+	label: {
+		fontSize: '16px',
+	},
+} );
+
+const EdgeCacheToggle = ( { onEdgeCacheToggle, checked, disabled } ) => {
+	const translate = useTranslate();
+	return (
+		<ToggleContainer>
+			<ToggleControl
+				disabled={ disabled }
+				checked={ checked }
+				onChange={ onEdgeCacheToggle }
+				label={ translate( 'Edge Cache' ) }
+			/>
+		</ToggleContainer>
+	);
+};
+
+export const CacheCard = ( {
 	disabled,
 	shouldRateLimitCacheClear,
 	clearAtomicWordPressCache,
-	isClearingCache,
+	isClearingWordpressCache,
 	siteId,
 	translate,
 } ) => {
+	const dispatch = useDispatch();
+	const showEdgeCache = config.isEnabled( 'edge-cache' );
+	const [ isEdgeCacheActive, setIsEdgeCacheActive ] = useState( false );
+	const { loading: getEdgeCacheLoading, data: edgeCacheActive } = useEdgeCacheQuery( siteId, {
+		enabled: showEdgeCache,
+	} );
```

Comment:
@kozer, would it be possible to remove the `useState` and use only the network response?
I see you are already using `queryClient.setQueryData`.

```suggestion

	const { loading: getEdgeCacheLoading, data: isEdgeCacheActive } = useEdgeCacheQuery( siteId, {
		enabled: showEdgeCache,
	} );
```

And then we could also remove this `useEffect`.

```jsx
	useEffect( () => {
		setIsEdgeCacheActive( edgeCacheActive );
	}, [ edgeCacheActive ] );
```